packages/expo-video-thumbnails/ios/EXVideoThumbnails/EXVideoThumbnailsModule.m

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXVideoThumbnails/EXVideoThumbnailsModule.h>
+#import <AVFoundation/AVFoundation.h>
+#import <AVFoundation/AVAsset.h>
+#import <UIKit/UIKit.h>
+
+NSString* const OPTIONS_KEY_QUALITY = @"quality";
+NSString* const OPTIONS_KEY_TIME = @"time";
+NSString* const OPTIONS_KEY_HEADERS = @"headers";
+
+@implementation EXVideoThumbnailsModule
+
+UM_EXPORT_MODULE(ExpoVideoThumbnails);
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+    _moduleRegistry = moduleRegistry;
+    _fileSystem = [moduleRegistry getModuleImplementingProtocol:@protocol(UMFileSystemInterface)];
+}
+
+UM_EXPORT_METHOD_AS(getThumbnail,
+                    sourceFilename:(NSString *)source
+                    options:(NSDictionary *)options
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject)
+{
+    NSURL *url = [NSURL URLWithString:source];
+    if ([url isFileURL]) {
+        if (!_fileSystem) {
+            return reject(@"E_MISSING_MODULE", @"No FileSystem module.", nil);
+        }
+        if (!([_fileSystem permissionsForURI:url] & UMFileSystemPermissionRead)) {
+            return reject(@"E_FILESYSTEM_PERMISSIONS", [NSString stringWithFormat:@"File '%@' isn't readable.", source], nil);
+        }
+    }
+    
+    long timeInMs = [(NSNumber* )options[OPTIONS_KEY_TIME] integerValue] ?: 0;
+    float quality = [(NSNumber* )options[OPTIONS_KEY_QUALITY] integerValue] ?: 1.0;
+    NSDictionary* headers = options[OPTIONS_KEY_HEADERS] ?: @{};
+    
+    
+    AVURLAsset *asset = [[AVURLAsset alloc] initWithURL:url options:@{@"AVURLAssetHTTPHeaderFieldsKey": headers}];
+    AVAssetImageGenerator *generator = [[AVAssetImageGenerator alloc] initWithAsset:asset];
+    generator.appliesPreferredTrackTransform = YES;
+    
+    NSError *err = NULL;
+    CMTime time = CMTimeMake(timeInMs, 1000);
+    
+    CGImageRef imgRef = [generator copyCGImageAtTime:time actualTime:NULL error:&err];
```

Comment:
Could you add handling for this `err`? Something similar to https://github.com/sjchmiela/react-native-video-thumbnailer/commit/148bf4600586bdbf56f6117cf60ad1af47bd15d8#diff-ce674e07779cf2c3b371f7ef1253ee18R36